tests/PathTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace OpenCFP\Test;
+
+use OpenCFP\Environment;
+use OpenCFP\Path;
+
+/**
+ * @covers \OpenCFP\Path
+ */
+class PathTest extends BaseTestCase
+{
+    /**
+     * @var Path
+     */
+    private $path;
+
+    public function setUp()
+    {
+        parent::setUp();
+        $this->path = new Path('/home/folder/base', Environment::testing());
+    }
+
+    /**
+     * @test
+     */
+    public function basePathReturnsBasePath()
+    {
+        $this->assertSame('/home/folder/base', $this->path->basePath());
+    }
+
+    /**
+     * @test
+     */
+    public function configPathReturnsConfigPath()
+    {
+        $this->assertSame(
+            '/home/folder/base/config/testing.yml',
+            $this->path->configPath()
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function configPathWorksForProdAndDev()
+    {
+        $prod = new Path('/home/folder/base', Environment::production());
+        $dev  = new Path('/home/folder/base', Environment::development());
+        $this->assertSame(
+            '/home/folder/base/config/production.yml',
+            $prod->configPath()
+        );
+        $this->assertSame(
+            '/home/folder/base/config/development.yml',
+            $dev->configPath()
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function uploadPathReturnsUploadPath()
+    {
+        $this->assertSame(
+            '/home/folder/base/web/uploads',
+            $this->path->uploadPath()
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function uploadPathReturnsUploadPathWhenProduction()
+    {
+        $path = new Path('/home/folder/base', Environment::production());
```

Comment:
Just food for thought, as I *used* to like setting up the system under test in `setUp()` - I stopped doing it because:

* I found tests became hard to understand (can't see everything in one place, need to go back to `setUp()`)
* when I end up with different scenarios, then I need to arrange individually anyway
* repetition in tests doesn't do any harm

Maybe this helps: https://stackoverflow.com/questions/6453235/what-does-damp-not-dry-mean-when-talking-about-unit-tests.